### PR TITLE
[ETL-490] Delete local copy of JSON after uploading to S3

### PR DESCRIPTION
**Pull request by philerooski**

### CODE REVIEW

File: tests/test_s3_to_json.py [hunk 1]
```diff
@@ -105,6 +105,7 @@ def test_write_healthkitv2samples_file_to_json_dataset(self, s3_obj, namespace,
                 dataset_identifier="HealthKitV2Samples",
                 metadata=sample_metadata["Metadata"],
                 workflow_run_properties=workflow_run_properties,
+                delete_upon_successful_upload=False,
```
Comment (rxu17) [inline tests/test_s3_to_json.py:108]: Do you think we need a test that tests that `os.remove` does get called as the default for `write_file_to_json_dataset` just to make sure that's always true?

**Review by rxu17 — APPROVED** [review]
LGTM! Just one comment

File: tests/test_s3_to_json.py (hunk 1 above)
Comment (philerooski) [reply tests/test_s3_to_json.py:108]: Yeah, I'll add it.

File: src/glue/jobs/s3_to_json.py [hunk 2]
```diff
@@ -21,7 +21,8 @@ def write_file_to_json_dataset(
         json_path: str,
         dataset_identifier: str,
         metadata: dict,
-        workflow_run_properties: dict) -> str:
+        workflow_run_properties: dict,
+        delete_upon_successful_upload=True) -> str:
```
Comment (thomasyu888) [inline src/glue/jobs/s3_to_json.py:25]: Nit: add type hint.

`bool = True`

**Review by thomasyu888 — APPROVED** [review]
💯 this is great to have! Thanks!

**Pull request merged**